Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -21,981 +21,660 @@
-    char* pPtr = (char*)img->Backing()->LockImageData();
-    CGImageDataProvider* ret = [[CGImageDataProvider alloc] initWithBytesNoCopy:pPtr
-                                                                         length:img->Backing()->Height() * img->Backing()->BytesPerRow()
-                                                                   freeWhenDone:FALSE];
-    ret->_img = img;
-
-    return ret;
-}
+CGColorRenderingIntent CGImageGetRenderingIntent(CGImageRef image) {
+  if (!image) {
+    return kCGRenderingIntentDefault;
+  }
 
-void* _CGImageGetData(CGImageRef img) {
-    return img->Backing()->StaticImageData();
+  return image->RenderingIntent();
 }
 
 /**
  @Status Interoperable
 */
-CGColorSpaceRef CGImageGetColorSpace(CGImageRef img) {
-    // TODO: Consider caching colorspaceRef in CGImageRef
-    CGColorSpaceRef ret = (CGColorSpaceRef) new __CGColorSpace(img->Backing()->ColorSpaceModel());
-
-    return ret;
+bool CGImageGetShouldInterpolate(CGImageRef image) {
+  RETURN_FALSE_IF(!image);
+  return image->Interpolate();
 }
 
 /**
  @Status Interoperable
 */
-size_t CGImageGetBitsPerPixel(CGImageRef img) {
-    if (!img) {
-        TraceWarning(TAG, L"CGImageGetBitsPerPixel: nil!");
-        return 0;
-    }
-
-    return (img->Backing()->BytesPerPixel() << 3);
+bool CGImageIsMask(CGImageRef image) {
+  RETURN_FALSE_IF(!image);
+  return image->IsMask();
 }
 
 /**
  @Status Interoperable
 */
-size_t CGImageGetBitsPerComponent(CGImageRef img) {
-    if (!img) {
-        TraceWarning(TAG, L"CGImageGetBitsPerComponent: nil!");
-        return 0;
-    }
-
-    return (img->Backing()->BitsPerComponent());
+CGColorSpaceRef CGImageGetColorSpace(CGImageRef img) {
+  RETURN_NULL_IF(!img);
+  return img->ColorSpace();
 }
 
 /**
  @Status Interoperable
 */
 size_t CGImageGetWidth(CGImageRef img) {
-    if (!img)
-        return 0;
-
-    return img->Backing()->Width();
+  RETURN_RESULT_IF_NULL(img, 0);
+  return img->Width();
 }
 
 /**
  @Status Interoperable
 */
 size_t CGImageGetHeight(CGImageRef img) {
-    if (!img)
-        return 0;
-
-    return img->Backing()->Height();
+  RETURN_RESULT_IF_NULL(img, 0);
+  return img->Height();
 }
 
 /**
  @Status Interoperable
 */
 void CGImageRelease(CGImageRef img) {
-    CFRelease((id)img);
+  RETURN_IF(!img);
+  CFRelease((CFTypeRef)img);
 }
 
 /**
  @Status Interoperable
 */
 CGImageRef CGImageRetain(CGImageRef img) {
-    CFRetain((id)img);
-    return img;
+  RETURN_NULL_IF(!img);
+  CFRetain((CFTypeRef)img);
+  return img;
 }
 
 /**
  @Status Interoperable
 */
 CGBitmapInfo CGImageGetBitmapInfo(CGImageRef img) {
-    int ret;
-
-    if (img) {
-        ret = img->Backing()->BitmapInfo();
-    } else {
-        TraceWarning(TAG, L"CGImageGetBitmapInfo: Null CGImageRef!");
-        ret = 0;
-    }
-
-    return ret;
+  RETURN_RESULT_IF_NULL(img, 0);
+  return img->BitmapInfo();
 }
 
 /**
- @Status Interoperable
+ @Status Caveat
+ @Notes decode parameter not supported and must be nullptr.
```

Comment:
The apple documentation makes clear of this. 

---
In reply to: [84548940](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84548940) [](ancestors = 84548940)